### ISSUE TITLE
Special global namespace does not parse properly

### ISSUE THREAD

**Issue by kitsonk** [dsherret/ts-simple-ast#438]
**Describe the bug**
Version: 14.4.3

The special `global` module/namespace does not parse properly, and you cannot author it properly.

The following is valid TypeScript in an ambient context:

```ts
declare global {
  const foo: string;
}
```

It is properly a module node, and `ts-simple-ast` parses it with the appropriate kind and assigns it a namespace declaration, but the methods on the node don't return as expected.

**To Reproduce**

For example the following code:

```ts
const globalNamespace = sourceFile.getNamespaceOrThrow("global");
console.log("hasDeclareKeyword:", globalNamespace.hasDeclareKeyword());
console.log("hasDefaultKeyword:", globalNamespace.hasDefaultKeyword());
console.log("hasExportKeyword:", globalNamespace.hasExportKeyword());
console.log("hasModuleKeyword:", globalNamespace.hasModuleKeyword());
console.log("hasNamespaceKeyword:", globalNamespace.hasNamespaceKeyword());
```

Will output:

```
hasDeclareKeyword: true
hasDefaultKeyword: false
hasExportKeyword: false
hasModuleKeyword: true
hasNamespaceKeyword: false
```

It would be expected that `hasModuleKeyword` is `false` and `hasNamespaceKeyword` is `false`.

When attempting to create, the following code:

```ts
const globalGlobalsModule = globalsModule.addNamespace({
  name: "global",
  hasModuleKeyword: false
});

globalGlobalsModule.setHasNamespaceKeyword(false);
```

Will still generate:

```ts
namespace global {
}
```

Declaring `module global` or `namespace global` does not have the same meaning to TypeScript and it does not add the types to the global namespace.

**Expected behavior**

The ability to properly parse and generate the special `global` namespace.

**Comment by dsherret** [comment]
I'm thinking I'm going to make some breaking changes here to properly support this scenario. Let me know your thoughts.

This patterns off `VariableDeclarationKind`.

Deprecating:

* `.setHasNamespaceKeyword`
* `.setHasModuleKeyword`
* `hasModuleKeyword` and `hasNamespaceKeyword` in `NamespaceDeclarationStructure`

Will add enum:

```ts
enum NamespaceDeclarationKind {
    Namespace: "namespace",
    Module: "module",
    Global: "global"
}
```

Adding:

* `setDeclarationKind(kind: NamespaceDeclarationKind)`
* `getDeclarationKind(): NamespaceDeclarationKind;`
* `NamespaceDeclarationStructure` - `declarationKind: NamespaceDeclarationKind`

Perhaps the following should also be deprecated, but I don't see any harm in leaving them:

* `.hasNamespaceKeyword`
* `.hasModuleKeyword`

**Comment by kitsonk** [comment]
That certainly would meet my requirements. 👍

**Comment by dsherret** [comment]
@kitsonk cool. Everything is available now in 16.0.0. Let me know if you run into any problems.
